### ISSUE TITLE
Best Epoch in Training

### ISSUE THREAD

**Issue by aseprohman** [ultralytics/yolov5#5322]
## ❔Question
Hello @glenn-jocher et all, 

I've tried to train yolov5 model with dataset (epoch:200,dataset:3300 images, class:1, label:>11000) and the result after the training was finished I checked the accuracy of the model (last.pt) to the real object sometimes lower than epoch50.pt or epoch100.pt and I have checked the validation loss graph and have not overfitted. Has anyone else experienced anything like this? Graphically, the accuracy of the last model should have the highest level of accuracy.

**Comment by ChinaRush** [comment]
the default traning epoch is 300, only the best.pt can get the best result of your task. for your question,  your model haven't converged ,so you need to adjust the training epoch to 300 or much bigger

**Comment by aseprohman** [comment]
I can't train dataset less than 300 epoch for get the best result ? I have only single class

**Comment by ChinaRush** [comment]
you can see your traning curve in tensorborad or W&B , when your training started converging, the corresponding epoch is your dataset  need to be trained if you want get best result. 300 is just a baseline , for  more large datasets, we even need to train 400, 600 or even more epochs until  the model overfitted.

**Comment by ghost** [comment]
After several weeks with yolov5, I found that you have to take a three-way approach for training: 

- less than 10 epochs to see if anything is wrong with your labels, bounding boxes, yaml etc. (extremely low mAP, like 0.00 something)
- 100 ep. to get meaningful results (mAP > 0.9, most test images will pass, for us that's the sweet spot for evaluating synthetic data)
- more than 200 ep. to get something usable

If you don't get good results after 300 ep. for just one class I would suspect an issue with the training set.

**Comment by aseprohman** [comment]
nice info, many thanks :) 
how about validation set ? I see in coco128 validation set is same with the train set, should I make validation set data is realy different from train set ? is it affect for model accuracy or just for model overfitted check only ?

**Comment by glenn-jocher** [comment]
@aseprohman COCO128 is just a toy dataset to verify error free overfitting is possible before committing to training custom datasets. For training and deployment of real-life products you want to have a separate val set. See https://en.wikipedia.org/wiki/Training,_validation,_and_test_sets

**Comment by aseprohman** [comment]
thanks all :)
I understand better
